tests/vmi_lifecycle_test.go

```diff
@@ -630,6 +631,61 @@ var _ = Describe("VMIlifecycle", func() {
 
 		})
 
+		Context("with node feature discovery", func() {
+
+			var options metav1.GetOptions
+			var cfgMap *k8sv1.ConfigMap
+			var originalData string
+
+			BeforeEach(func() {
+				options = metav1.GetOptions{}
+				cfgMap, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Get("kubevirt-config", options)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				originalData = cfgMap.Data[featuregates.FEATURE_GATES_KEY]
+				cfgMap.Data[featuregates.FEATURE_GATES_KEY] = featuregates.CPUNodeDiscoveryGate
+				_, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Update(cfgMap)
```

Comment:
The tests are racy. We will have to improve the detection if virt-controller already received the config-map change. I don't want to make this too complicated for you right now, but one proper solution would be to expose the config-map feature gates on the already existing healthz endpoint of virt-controller. Then this endpoint can be queried and as soon as it reports the right flags, we know that all subsequent calls inside the virt-controller code will see the exact same values.

Reply:
If you e.g. wait half a second, put a fixme here and add the negative test, then this should work for now good enough. Otherwise you will have to change a lot. @davidvossel thoughts ?

Reply:
add a sleep.